### PR TITLE
Feature: Resource User Notification Rule

### PR DESCRIPTION
**Pull request by jaceq**
I've added new resource User Notification Rule.

Any comments / hints / feedback are welcome.

### CODE REVIEW

**Comment by jaceq** [comment]
@heimweh @gordonbondon  Could anybody take a look here, please?

**Comment by gordonbondon** [comment]
@jaceq  hi! I'm not the maintainer of this provider, so, unfortunately, I won't be able to help you to get this merged.

**Comment by jaceq** [comment]
Hello? Any update on that?

File: pagerduty/resource_pagerduty_user_notification_rule_test.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  user_id = "${pagerduty_user.foo.id}"
+  type    = "email_contact_method"
+  address = "foo-1@bar.com"
+  label   = "Work"
+}
+
+resource "pagerduty_user_contact_method" "sms_contact_method" {
+  user_id      = "${pagerduty_user.foo.id}"
+  type         = "sms_contact_method"
+  address      = "1234567890"
```
Comment (pdecat) [inline pagerduty/resource_pagerduty_user_notification_rule_test.go]: When running acceptance tests, this is currently rejected by the Pagerduty API:

```
make testacc TEST=./pagerduty TESTARGS='-run=TestAccPagerDutyUserNotification -count=1'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./pagerduty -v -run=TestAccPagerDutyUserNotification -count=1 -timeout 120m
=== RUN   TestAccPagerDutyUserNotificationRule_import
--- FAIL: TestAccPagerDutyUserNotificationRule_import (7.12s)
    testing.go:538: Step 0 error: Error applying: 2 errors occurred:
                * pagerduty_user_contact_method.phone_contact_method: 1 error occurred:
                * pagerduty_user_contact_method.phone_contact_method: POST API call to https://api.pagerduty.com/users/PFKZX66/contact_methods failed 400 Bad Request. Code: 2001, Errors: [Phone number does not seem to be valid Please contact support for assistance], Message: Invalid Input Provided


                * pagerduty_user_contact_method.sms_contact_method: 1 error occurred:
                * pagerduty_user_contact_method.sms_contact_method: POST API call to https://api.pagerduty.com/users/PFKZX66/contact_methods failed 400 Bad Request. Code: 2001, Errors: [Phone number does not seem to be valid Please contact support for assistance], Message: Invalid Input Provided




=== RUN   TestAccPagerDutyUserNotificationRuleContactMethod_Basic
--- FAIL: TestAccPagerDutyUserNotificationRuleContactMethod_Basic (6.78s)
    testing.go:538: Step 0 error: Error applying: 2 errors occurred:
                * pagerduty_user_contact_method.sms_contact_method: 1 error occurred:
                * pagerduty_user_contact_method.sms_contact_method: POST API call to https://api.pagerduty.com/users/PCOOWTG/contact_methods failed 400 Bad Request. Code: 2001, Errors: [Phone number does not seem to be valid Please contact support for assistance], Message: Invalid Input Provided


                * pagerduty_user_contact_method.phone_contact_method: 1 error occurred:
                * pagerduty_user_contact_method.phone_contact_method: POST API call to https://api.pagerduty.com/users/PCOOWTG/contact_methods failed 400 Bad Request. Code: 2001, Errors: [Phone number does not seem to be valid Please contact support for assistance], Message: Invalid Input Provided




FAIL
FAIL    github.com/terraform-providers/terraform-provider-pagerduty/pagerduty   13.911s
FAIL
make: *** [GNUmakefile:17: testacc] Error 1
```

Using `8005551234` works:

```
make testacc TEST=./pagerduty TESTARGS='-run=TestAccPagerDutyUserNotification -count=1'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./pagerduty -v -run=TestAccPagerDutyUserNotification -count=1 -timeout 120m
=== RUN   TestAccPagerDutyUserNotificationRule_import
--- PASS: TestAccPagerDutyUserNotificationRule_import (11.57s)
=== RUN   TestAccPagerDutyUserNotificationRuleContactMethod_Basic
--- PASS: TestAccPagerDutyUserNotificationRuleContactMethod_Basic (19.69s)
PASS
ok      github.com/terraform-providers/terraform-provider-pagerduty/pagerduty   31.272s
```


```suggestion
  address      = "8005551234"
```

File: pagerduty/resource_pagerduty_user_notification_rule_test.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  country_code = "+1"
+  label        = "Work"
+}
+
+resource "pagerduty_user_contact_method" "phone_contact_method" {
+  user_id      = "${pagerduty_user.foo.id}"
+  type         = "phone_contact_method"
+  country_code = "+1"
+  address      = "1234567890"
```
Comment (pdecat) [inline pagerduty/resource_pagerduty_user_notification_rule_test.go]:
```suggestion
  address      = "8005551234"
```

**Comment by jaceq** [comment]
Hi @pdecat  I submitted this PR almost a year ago...
I have used a custom version of pagerduty provider since then until... we moved away from pagerduty  (literally 1-2 weeks ago).
Because of this it will be hard to find time to adjust this.
Let me know if you / someone can take this over or should I just close this PR.

Regards

File: pagerduty/import_pagerduty_user_notification_rule_test.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package pagerduty
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
```
Comment (pdecat) [inline pagerduty/import_pagerduty_user_notification_rule_test.go]:
```suggestion
	"github.com/hashicorp/terraform-plugin-sdk/terraform"
```

File: pagerduty/import_pagerduty_user_notification_rule_test.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
+package pagerduty
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
```
Comment (pdecat) [inline pagerduty/import_pagerduty_user_notification_rule_test.go]:
```suggestion
	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
```

File: pagerduty/resource_pagerduty_user_notification_rule.go [hunk 5]
```diff
@@ -0,0 +1,184 @@
+package pagerduty
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
```
Comment (pdecat) [inline pagerduty/resource_pagerduty_user_notification_rule.go]:
```suggestion
	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
```

File: pagerduty/resource_pagerduty_user_notification_rule_test.go [hunk 6]
```diff
@@ -0,0 +1,132 @@
+package pagerduty
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
```
Comment (pdecat) [inline pagerduty/resource_pagerduty_user_notification_rule_test.go]:
```suggestion
	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
```

File: pagerduty/resource_pagerduty_user_notification_rule_test.go [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+	"github.com/hashicorp/terraform/terraform"
```
Comment (pdecat) [inline pagerduty/resource_pagerduty_user_notification_rule_test.go]:
```suggestion
	"github.com/hashicorp/terraform-plugin-sdk/terraform"
```

File: pagerduty/resource_pagerduty_user_notification_rule_test.go [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+func testAccCheckPagerDutyUserNotificationRuleContactMethodConfig(methodType string) string {
+	return fmt.Sprintf(`
+resource "pagerduty_user_notification_rule" "foo" {
+  user_id                = "${pagerduty_user.foo.id}"
+  start_delay_in_minutes = 1
+  urgency                = "high"
+
+  contact_method {
```
Comment (pdecat) [inline pagerduty/resource_pagerduty_user_notification_rule_test.go]:
```suggestion
  contact_method = {
```

**Comment by pdecat** [comment]
@jaceq, I've opened a new PR based on yours https://github.com/terraform-providers/terraform-provider-pagerduty/pull/180

Thanks again!

**Comment by jaceq** [comment]
Thanks for taking this over @pdecat  I hope new PR will get merged in some close future :)

**Pull request closed without merging**